source/lib/gull/etcd.go

```diff
@@ -2,40 +2,58 @@ package gull
 
 import (
 	"fmt"
+	"net/url"
 
 	"github.com/franela/goreq"
 )
 
 type EtcdMigrationTarget struct {
 	EtcdHostUrl string
+	Environment string
 }
 
-func NewEtcdMigrationTarget(hostUrl string) *EtcdMigrationTarget {
+func NewEtcdMigrationTarget(hostUrl string, environment string) *EtcdMigrationTarget {
 	return &EtcdMigrationTarget{
 		EtcdHostUrl: hostUrl,
+		Environment: environment,
 	}
 }
 
 func (emt *EtcdMigrationTarget) Set(path string, value string) error {
 	if emt.EtcdHostUrl == "" {
 		return fmt.Errorf("EtcdMigrationTarget's EtcdHostUrl cannot be empty")
 	}
-	storageUrl := emt.EtcdHostUrl + path
+	storageUrl := emt.EtcdHostUrl + url.QueryEscape(path)
+	value = fmt.Sprintf("value=%v", url.QueryEscape(value))
+	fmt.Printf("Writing [%v]->[%v]\n", storageUrl, value)
 	response, err := goreq.Request{
-		Method: "PUT",
-		Uri:    storageUrl,
-		Body:   value,
+		Method:      "PUT",
+		Uri:         storageUrl,
+		Body:        value,
+		ContentType: "application/x-www-form-urlencoded",
 	}.Do()
 	if response != nil {
 		defer func() { _ = response.Body.Close() }()
```

Comment:
IIRC it was a habit I got into after running either govet or errcheck against the c2fo-go baseline. Speaking of that, I'm going to open a ticket to see if we can automatically run those two in travis.